src/shared/proto/tbots_robot_msg.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto2";
+
+import "shared/proto/robot_component_status.proto";
+
+message TbotsRobotMsg
+{
+    required uint32 robot_id                           = 1;
+    repeated ErrorCode error_code                      = 2;
+    required BreakBeamStatus break_beam_status         = 3;
+    required ChipperKickerStatus chipper_kicker_status = 4;
+    required MotorEncoderData motor_encoder_data       = 5;
+    required FirmwareStatus firmware_status            = 6;
+    required DribblerStatus dribbler_status            = 7;
+    required NetworkStatus network_status              = 8;
+    required PowerStatus power_status                  = 9;
+    required TemperatureStatus temperature_status      = 10;
```

Comment:
Thoughts on whether `required` fields should be `optional` instead? (https://github.com/UBC-Thunderbots/Software/pull/1274#pullrequestreview-396471163)

Reply:
Id vouch for optional everywhere for consistency sake

`NetworkStatus` for example has fields that are entirely optional, but the `network_status` msg itself is required?

Reply:
I'm fine with leaving _some_ things required. The reason is that compared to the corporate users of protobuf, we care relatively less about long-term backwards compatibility and we control everything ourselves.

But in the interest of flexibility and recommended style, I have no problems with most/all things being optional. I think we're safe to leave a few things that we are _confident_ would never change, like battery and cap voltage, but if this would make the structure weird then IMO it's not worth it.

Reply:
If something does not need to be optional it shouldn't be, as optional things incur more syntactic and semantic overhead for the developer working with them.

Reply:
turns out proto3 doesn't support required so we're stuck with optional